src/main/java/com/brightpath/learnify/domain/note/NoteService.java

```diff
@@ -79,22 +80,28 @@ public Note getNoteById(UUID uuid) {
     }
 
     @Transactional
-    public void updateBoardNoteContentPage(UUID uuid, Integer pageNumber, String body) {
-        boardNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body);
+    public void updateBoardNoteContentPage(UUID uuid, Integer pageNumber, String body, Integer version) {
+        int rowsAffected = boardNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body, version);
+        if (rowsAffected == 0) {
+            throw new ResourceUpdateConflictException(BOARD_NOTE_PAGE, uuid);
+        }
     }
 
     @Transactional
-    public void updateDocumentNoteContentPage(UUID uuid, Integer pageNumber, String body) {
-        documentNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body);
+    public void updateDocumentNoteContentPage(UUID uuid, Integer pageNumber, String body, Integer version) {
+        int rowsAffected = documentNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body, version);
```

Comment:
how this rows affected works?

Reply:
The sql update query returns how many records were updated. Normally we expect that one row was affected - if not then we assume that provided version is not actual to allow the update so we throw Confilct exception